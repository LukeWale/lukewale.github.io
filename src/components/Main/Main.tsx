import React from "react";

const Main = () => {
    return <div data-testid="app-main"></div>
}

export default Main;