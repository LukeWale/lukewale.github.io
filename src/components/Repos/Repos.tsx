import React from "react";
import GithubReposData from '../../data/repos.json';
import imagePlaceholder from '../../assets/images/placeholder-image.png';
import './Repos.css';

const Repos = () => {
    return (
        <div data-testid="repos">
            <h2 className="text-center text-4xl">Public Repos</h2>
            <div data-testid="repo-list" className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3">
                {
                    GithubReposData.map(singleData => {
                        let pagesURL = singleData.pagesUrl;
                        let username = singleData.userName;
                        return (
                            singleData.repos.map(repo => {
                                return (
                                    <div key={repo.githubUrl} className="text-center">
                                        <img src={imagePlaceholder} alt="" data-testid="repo-image"/>
                                        <h3 data-testid="repo-name">{repo.name}</h3>
                                        <a data-testid="repo-github-link" className="bg-primary p-5 mb-[10px] block rounded-[10px] text-white font-semibold" href={`https://github.com/${username}/${repo.githubUrl}`} target="_blank" rel="noreferrer" >Repo</a>
                                        <a data-testid="repo-view-link" className="bg-primary p-5 mb-[10px] block rounded-[10px] text-white font-semibold" href={`${pagesURL}/${repo.githubPage}`} target="_blank" rel="noreferrer">View</a>
                                    </div>
                                )
                            })
                        )
        
                        
                            
                        })
                }
            </div>
        </div>
    )
}

export default Repos;