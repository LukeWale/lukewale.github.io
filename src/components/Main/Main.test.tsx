import React from "react";
import { cleanup, render, screen } from '@testing-library/react';
import Main from './Main';
import Repos from '../Repos/Repos';

describe('Main', () => {
    
    afterEach(cleanup);
    
    it('should render correctly', () => {
        const { asFragment } = render(<Main />);

        expect(asFragment()).toMatchSnapshot();
    })

    it('has a container of main', () => {
        render(<Main />);

        expect(screen.getByRole('main')).toBeInTheDocument();
    });

    it('has a container element', () => {
        const {container} = render(<Main />);

        // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
        const containers = container.getElementsByClassName('container');

        expect(containers.length).toBe(1);
    });

    it('has a h1 tag', () => {
        render(<Main />);

        const heading = screen.getByRole('heading', { level: 1 });

        expect(heading).toBeInTheDocument();
    });

    it('renders the repos component', () => {
        render(<Main />);

        const element = screen.getByTestId('repos');

        expect(element).toBeInTheDocument();
    });

    it('has a logo animation gif', () => {
        render(<Main />);

        const logoImage = screen.getByTestId('introImage');

        expect(logoImage).toBeInTheDocument();
    });


});