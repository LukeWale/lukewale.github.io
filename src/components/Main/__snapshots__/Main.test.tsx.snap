// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
<DocumentFragment>
  <main
    class="pt-24"
    data-testid="app-main"
  >
    <div
      class="container p-6"
    >
      <div
        class="text-center"
      >
        <img
          alt=""
          class="mx-auto"
          data-testid="introImage"
          src="logo-animation.gif"
        />
      </div>
      <div
        class="text-center py-5"
      >
        <h1>
          About me
        </h1>
        <p>
          I'm interested in making things happen... I've done all the static sites, I've done the modifications to others systems, I've done the custom wordpress plugins... now I'm excited for building my own things. After working for multiple employers for long periods of time I'd love to be able to have my own portfolio of things I've enjoyed doing... If you can't see things here it is likely I have them set to private repos, don't worry!
        </p>
        <p>
          I'm currently focusing on JS, React, APIs, etc.
        </p>
        <p>
          Over time I will be making little snippets public, so check below for updates!
        </p>
      </div>
      <div
        data-testid="repos"
      >
        <h2
          class="text-center text-4xl"
        >
          Public Repos
        </h2>
        <div
          class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3"
          data-testid="repo-list"
        >
          <div
            class="text-center"
          >
            <img
              alt=""
              data-testid="repo-image"
              src="placeholder-image.png"
            />
            <h3
              data-testid="repo-name"
            >
              JS QR Code Generator
            </h3>
            <a
              class="bg-primary p-5 mb-[10px] block rounded-[10px] text-white font-semibold"
              data-testid="repo-github-link"
              href="https://github.com/LukeWale/js-qr-code-generator"
              rel="noreferrer"
              target="_blank"
            >
              Repo
            </a>
            <a
              class="bg-primary p-5 mb-[10px] block rounded-[10px] text-white font-semibold"
              data-testid="repo-view-link"
              href="https://lukewale.github.io/js-qr-code-generator"
              rel="noreferrer"
              target="_blank"
            >
              View
            </a>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
